src/components/AddressModal/AddressModal.js

```diff
@@ -0,0 +1,121 @@
+    const [formDetails, setFormDetails] = useState(
+        editAddress ?? {
+            firstName: "",
+            mobileNumber: "",
+            pinCode: "",
+            home: "",
+            state: "",
+            country: "",
+        })
+
+    const dummyDataHandler = () => {
+        setFormDetails(addressDetails);
+    }
+    const inputHandler = (e, field) => {
+        setFormDetails((pre) => ({
+            ...pre, [field]: e.target.value
+        }))
+    }
+    const formHandler = (e) => {
+        e.preventDefault()
+        addressDispatch({
+            type: "UPDATE_ADDRESS",
+            payload: !editAddress
+                ? { _id: uuid(), ...formDetails }
+                : { isEdit: true, addressList: formDetails },
+        })
+        setShowModal(false);
+    }
+    return ReactDOM.createPortal(
+        <div className=" modal-active modal-container">
+            <div className={`${styles.dialog} dialog`}>
+                <span className={`${styles.modal_header} modal-title fn-size-l`}>Add new address
+                    <button className='btn btn-icon-only' onClick={() => setShowModal(false)}>
+                        <span className="material-icons">
+                            close
+                        </span>
+                    </button>
+                </span>
+                <form onSubmit={formHandler}>
+                    <input
+                        placeholder="Enter your name"
+                        className="input-field"
+                        type="text"
+                        id="firstName"
+                        value={formDetails.firstName}
+                        required
+                        onChange={(e) => inputHandler(e, "firstName")}
+                    />
+                    <input
+                        placeholder="Enter mobile number"
+                        className="input-field"
+                        type="number"
+                        id="mobileNumber"
+                        value={formDetails.mobileNumber}
+                        required
+                        onChange={(e) => inputHandler(e, "mobileNumber")}
+                    />
+                    <input
+                        placeholder="Pin code"
+                        className="input-field"
+                        type="number"
+                        id="pinCode"
+                        value={formDetails.pinCode}
+                        required
+                        onChange={(e) => inputHandler(e, "pinCode")}
+                    />
+                    <input
+                        placeholder="Enter house number, flat number"
+                        className="input-field"
+                        type="text"
+                        id="home"
+                        value={formDetails.home}
+                        required
+                        onChange={(e) => inputHandler(e, "home")}
+                    />
+                    <input
```

Comment:
Not mandatory, just a suggestion- you can create a generic component for input as the code is repeating. This way you will have a reusable component and will have to write less code.
You can do it later.